src/Lykke.Job.MtConsistencyChecker.AzureRepositories/AccountMarginEventReportRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using AzureStorage;
+using AzureStorage.Tables;
+using Common.Log;
+using Lykke.Job.MtConsistencyChecker.AzureRepositories.Entities;
+using Lykke.Job.MtConsistencyChecker.Contract;
+using Lykke.Job.MtConsistencyChecker.Core.Repositories;
+using Lykke.Job.MtConsistencyChecker.Core.Settings.JobSettings;
+using Lykke.SettingsReader;
+using Microsoft.WindowsAzure.Storage.Table;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Lykke.Job.MtConsistencyChecker.AzureRepositories
+{
+    public class AccountMarginEventReportRepository : IAccountMarginEventReportRepository
+    {
+        private readonly INoSQLTableStorage<AccountMarginEventReportEntity> _tableStorage;
+
+        public AccountMarginEventReportRepository(IReloadingManager<string> connectionString, ILog log)
+        {
+            _tableStorage = AzureTableStorage<AccountMarginEventReportEntity>.Create(connectionString,
+                "AccountMarginEventsReports", log);
+        }
+
+        public async Task<IEnumerable<IAccountMarginEventReport>> GetAsync(DateTime? dtFrom, DateTime? dtTo)
+        {
+            var partitionKeys = await GetPartitionKeys();
+            return (await _tableStorage.WhereAsync(partitionKeys, dtFrom ?? DateTime.MinValue, dtTo ?? DateTime.MaxValue, ToIntervalOption.IncludeTo))
+                .OrderByDescending(item => item.EventTime);
+        }
+
+        private async Task<IEnumerable<string>> GetPartitionKeys()
+        {
+            System.Collections.Concurrent.ConcurrentDictionary<string, byte> partitionKeys = new System.Collections.Concurrent.ConcurrentDictionary<string, byte>();
+            await _tableStorage.ExecuteAsync(new TableQuery<AccountMarginEventReportEntity>(), entity =>
+            {
+                foreach (var et in entity.Select(m => m.PartitionKey))
+                    partitionKeys.TryAdd(et, 0);
```

Comment:
Why do you use ConcurrentDictionary ?
If there's no need to save any value it's better to use ConcurrentBag<>.

Reply:
Solved